includes/constants/class-base-constant.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/**
+ * Class Base_Constant
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Constants;
+
+use ReflectionClass;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+
+/**
+ * Base constant class to hold common logic for all constants.
+ */
+abstract class Base_Constant {
```

Comment:
The reason why I put it as a base const since we're only using this for constants, and if you look from a different perspective, the enums are constants with steroids, and I wasn't sure where to put enum class since it felt wired to put it inside const folder with "Enum" name.